kebechet/config.py

```diff
@@ -93,6 +98,10 @@ def run_webhook(cls, payload: dict):
 
     @classmethod
     def run_url(cls, url: str, service: str, parsed_payload: dict, tls_verify: bool):
+        for instance in cls._CUSTOM_DEPLOYMENTS.keys():
+            if instance in url:
+                service = cls._CUSTOM_DEPLOYMENTS[instance]
+
```

Comment:
Maybe we could make `service` optional so that we can do some error checking if it is `None` after going through this for loop.

Reply:
Umm maybe this way the function demand's service type, and it it's not a valid type it error's out anyway.

Reply:
But if we are using a custom deployment what if the `service` isn't set?

Reply:
We have gitlab that is hosted by us (a custom deployment). Using "gitlab" service type should trigger logic for contacting gitlab instance by using GitLab specific API endpoints (the public one by default) and the service url, if stated, should adjust where to find the service.

Reply:
Then it resorts to the default service passed to the function, that way we support normal `gitlab` service types, and only resort to the custom instance's in the case where the URL contains, custom deployments we support.

Reply:
> We have gitlab that is hosted by us (a custom deployment). Using "gitlab" service type should trigger logic for contacting gitlab instance by using GitLab specific API endpoints (the public one by default) and the service url, if stated, should adjust where to find the service.

That's what it is doing now, it adjusts to custom deployments in case if the instance URL matches the type's we support.